sandpack-react/src/utils/array.ts

```diff
@@ -0,0 +1,7 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export const shallowEqual = (a: any[], b: any[]): boolean => {
+  if (a.length !== b.length) return false;
+
+  return a.every((item, index) => item === b[index]);
```

Comment:
Nit: this always traverses all the items. A regular for loop would be able to exit early on the first mismatches. Doesn't matter for small arrays though.

Reply:
Actually wait, I'm wrong. I think it _would_ exit early. So nvm.

Reply:
If I'm doing this right, for-loop is still the fastest option compared to the `some` or `every` implementation: 
![Screenshot 2022-05-26 at 17 33 08](https://user-images.githubusercontent.com/4838076/170532718-e83f64e8-1dde-4d35-b214-3e8b163df833.png)

Reply:
As far as I know, `some` or `every` exit as soon as a predicate match, but I couldn't find this in any documentation. If you have it, please let me know

Reply:
loops ftw

Reply:
Yes, it does an early exit ("If _testResult_ is **false**, return **false**."). https://tc39.es/ecma262/#sec-array.prototype.every